src/main/java/com/mparticle/kits/AppboyKit.java

```diff
@@ -261,20 +279,7 @@ public void setUserAttribute(String key, String value) {
             if (key.startsWith("$")) {
                 key = key.substring(1);
             }
-            if (value.equalsIgnoreCase("true") || value.equalsIgnoreCase("false")) {
-                user.setCustomUserAttribute(key, Boolean.parseBoolean(value));
-            } else {
-                try {
-                    double doubleValue = Double.parseDouble(value);
-                    if ((doubleValue % 1) == 0) {
-                        user.setCustomUserAttribute(key, Integer.parseInt(value));
-                    } else {
-                        user.setCustomUserAttribute(key, doubleValue);
-                    }
-                } catch (NumberFormatException nfe) {
-                    user.setCustomUserAttribute(key, value);
-                }
-            }
+            userAttributeSetter.parseValue(key, value);
```

Comment:
I think you want to check `enableDetectionType` here

Reply:
updated this so the `UserAttributeSetter` accepts `enableTypeDetection` as a constructor argument